web/src/main/java/se/inera/intyg/webcert/web/csintegration/message/ProcessIncomingMessageService.java

```diff
@@ -65,8 +70,15 @@ public SendMessageToCareResponseType process(SendMessageToCareType sendMessageTo
             isAnswer
         );
 
-        publishCertificateStatusUpdateService.publish(certificate, getEventType(questionType, isAnswer));
-        sendMailNotificationForReceivedMessageService.send(sendMessageToCare, certificate);
+        final var shouldReciveMailNotifications = shouldReciveMailNotifications(certificate);
+
+        if (unitIsIntegrated(certificate) && !shouldReciveMailNotifications) {
```

Comment:
You could probably simplify this a little bit. For example... You check first if there should be mail-notifications and if not do publish. Would be easier to understand the code and only one thing will be executed. Ex:

`
if (!unitIsIntegrated(certificate) || shouldReciveMailNotifications) {
    sendMailNotificationForReceivedMessageService.send(sendMessageToCare, certificate);
} else {
   publishCertificateStatusUpdateService.publish(certificate, getEventType(questionType, isAnswer));
}
`